### DISCUSSION TITLE
vite에서 env파일 사용시 주의할 점

### DISCUSSION THREAD

**Discussion by kimisadev27** [Dev-FE-1/Toy_Project_2_Team1#28] [Trouble Shooting]
firebase의 접속정보를 숨기고 환경변수로 처리하고자  dotenv 패키지를 설치해 사용하려고 했다.

그런데 vite의 경우 dotenv패키지없이 사용이 가능했고, 
기존 cra에서 사용법과 약간의 차이가 있었다.

CRA에서는 process.env.변수명 으로 접근이 가능했다면
VITE에서는 import.meta.env.변수명 으로 접근이 가능했다.

또 변수명에서도 차이가 있었는데
CRA는 변수명 앞에 REACT_APP_ 을 붙여야 접근가능 했다면
VITE에서는 VITE_를 붙여야 접근가능하다.

이점 참고하여 코드를 작성하도록 하자.

<img width="453" alt="image" src="https://github.com/user-attachments/assets/d9cc1886-2cd9-468d-b893-55086f3a45d4">

**Comment by badaclockA** [comment]
vite 공식문서에서도 명시된 내용입니다.
추가적으로 import.meta.env를 사용해야만 env 내용을 import 할 수 있어요.

https://vitejs.dev/guide/env-and-mode
![image](https://github.com/user-attachments/assets/1db9ea5e-d2c2-4540-a875-627a7fbe29d9)

**Comment by badaclockA** [comment]
추가적으로 typescript를 사용하신다면 vite-env.d.ts  파일에서 인터페이스와 vite/client 명시해야만 ts 경고 메시지가 뜨지 않습니다.
https://vitejs.dev/guide/env-and-mode#intellisense-for-typescript

vite-env.d.ts
```js
/// <reference types="vite/client" />

interface ImportMetaEnv {
  readonly VITE_APP_TITLE: string
  // more env variables...
}

interface ImportMeta {
  readonly env: ImportMetaEnv
}
```
